packages/controlled-form/src/lib/Input.tsx

```diff
@@ -41,14 +55,15 @@ export const ControlledInput = ({
         value,
         shouldUnregister,
         deps,
+        ...rules,
       }}
       shouldUnregister={shouldUnregister}
       render={({ field, fieldState: { error } }) => (
         <Input
           {...rest}
           {...field}
           error={!!error}
-          required={typeof required === 'object' ? required.value : !!required}
+          required={!!required}
```

Comment:
This does not work in all cases. See https://github.com/Availity/element/pull/658